feat(mobile): Pressable Text addition to the texts across Forum and Quiz pages

anilanlar opened the pull request:
Added word meaning search functionality to the following pages and components:

- forumpostpage
- quizdetails
- quizquestion
- quizresults
- quizreview
- postcard
- quizcard

### Forum 
<img width="333" alt="image" src="https://github.com/user-attachments/assets/82fa42b5-5ab8-4de4-885b-80c14ceb8747" />

### Quiz
<img width="334" alt="image" src="https://github.com/user-attachments/assets/b16b3721-0c8d-4e92-aa86-42c98e87a897" />

YavizGuldalf reviewed (changes requested):
This PR includes a lot of work, but I feel like it's still missing some features. First of all, the comments in the forum do not use the pressable text component. Secondly, the text in the forums are now centered, that is not a good look at all, I think we should fix that. Finally, adding a space after each word is a good idea, but it makes the dictionary meanings/examples look bad, the styling needs to be adjusted for those.

anilanlar commented:
I have 

- Added new comment functionality is added by means of a comment modal.
- fixed the appearance issue with the Bookmarked Words component.
- aligned Post Card titles to the left.
- corrected addComment api request.

YavizGuldalf reviewed (approved):
The changes look good in the UI, thank you @anilanlar. All the things I have asked for have been fixed.
